### PR TITLE
Use function to catch exceptions API calls

### PR DESCRIPTION
**Pull request by tbille**
# Summary

- Use function `_handle_errors` to catch exceptions

# QA

- `./run`
- Post a new username
- Post a new agreement
- Should work as usual

### CODE REVIEW

File: modules/publisher/views.py [hunk 1]
```diff
@@ -116,22 +116,12 @@ def post_agreement():
...
         try:
             api.post_agreement(flask.session, True)
-        except ApiTimeoutError as api_timeout_error:
-            flask.abort(504, str(api_timeout_error))
-        except ApiConnectionError as api_connection_error:
-            flask.abort(502, str(api_connection_error))
-        except ApiResponseDecodeError as api_response_decode_error:
-            flask.abort(502, str(api_response_decode_error))
         except ApiResponseErrorList as api_response_error_list:
             codes = [error['code'] for error in api_response_error_list.errors]
```
Comment (bartaz) [inline modules/publisher/views.py]: Why is not `_handle_errors_list` used here as well?

File: modules/publisher/views.py [hunk 2]
```diff
@@ -149,20 +139,10 @@ def post_account_name():
...
         try:
             api.post_username(flask.session, username)
-        except ApiTimeoutError as api_timeout_error:
-            flask.abort(504, str(api_timeout_error))
-        except ApiConnectionError as api_connection_error:
-            flask.abort(502, str(api_connection_error))
-        except ApiResponseDecodeError as api_response_decode_error:
-            flask.abort(502, str(api_response_decode_error))
         except ApiResponseErrorList as api_response_error_list:
             errors = errors + api_response_error_list.errors
```
Comment (bartaz) [inline modules/publisher/views.py]: Why is not `_handle_errors_list` used here as well?

File: modules/publisher/views.py (hunk 1 above)
Comment (tbille) [reply modules/publisher/views.py]: `_handle_errors_list` handles known errors and returns an `abort` while for those functions I don't too.

File: modules/publisher/views.py (hunk 2 above)
Comment (tbille) [reply modules/publisher/views.py]: same answer

**Comment by bartaz** [comment]
Wasn't able to test any API errors, but standard flow seems to still work.

**Pull request merged**